src/pages/docs/creators/common/requestor-provider-interaction.md

```diff
@@ -0,0 +1,293 @@
+
+```bash
+ProposalRejected {
+  id: 'R-dfb4c352fe12227a22cdedcac146d6b0d56f9dd69ef14da098106f58e20707bf',
+  parentId: 'R-2cad4a7562beaf13491d554c7536e2d467fc692c447de053e1a5b5aa0c422470',
+  reason: 'No capacity available. Reached Agreements limit: 1'
+}
+```
+
+### Agreement
+
+Once all the conditions are agreed upon, the requestor will create an `agreement` based on the latest proposals and send it to the Provider.
+The `agreement` is a set of eventually agreed conditions that will define the engagement between the requestor and a provider.
+
+```bash
+AgreementCreated {
+  id: '0ecd3882e3460d47ff98cb5d2abb9b0ad055ae8be79074614d03102b8a81e45f',
+  provider: {
+    id: '0x3d34d91eb4ab14b9f4fd8ac8985f82cc9eb28a92',
+    name: 'UK1-9.h',
+    walletAddress: '0x3d34d91eb4ab14b9f4fd8ac8985f82cc9eb28a92'
+  },
+  validTo: '2024-03-29T15:31:56.366Z',
+  proposalId: 'R-794d1d8e5230e9707f44cae4f6d49797a0278db4269890c66bd7e91f1f4531f6'
+}
+```
+
+Provider confirms the agreement.
+
+```bash
+AgreementConfirmed {
+  id: '0ecd3882e3460d47ff98cb5d2abb9b0ad055ae8be79074614d03102b8a81e45f',
+  provider: {
+    id: '0x3d34d91eb4ab14b9f4fd8ac8985f82cc9eb28a92',
+    name: 'UK1-9.h',
+    walletAddress: '0x3d34d91eb4ab14b9f4fd8ac8985f82cc9eb28a92'
+  }
+}
+```
+
+Note that the requestor may have agreements with multiple providers at the same time and distribute its task in parallel to all of them.
+
+### Activity
+
+Once the agreement is confirmed by both parties, we can start the deployment of our image to run a VM. At first, the `activity` is created. You can imagine an activity as a representation of a VM instance run on a provider. In theory, you may have more than one activity under the same agreement, but at this moment typical providers offer a single one at the same time.
+
+```bash
+ActivityCreated {
+  id: 'b284fcf996ed4b8da7cd1012233d23a1',
+  agreementId: '0ecd3882e3460d47ff98cb5d2abb9b0ad055ae8be79074614d03102b8a81e45f'
+}
+```
+
+All the time we can monitor activity status to know if the VM on the provider node is still available and running. Now it is initialized and ready for our commands.
+
+```bash
+ActivityStateChanged { id: 'b284fcf996ed4b8da7cd1012233d23a1', state: 'Initialized' }
+```
+
+### Scripts
+
+In the task model, the chunk of work is defined as a task (function). A task is a set of commands that the requestor wants to execute to obtain desired results.
+Internally all the commands are organized as `scripts`, that are sent and executed on a provider. There are several commands (depending on the time of type of the runtime utilized) available for the requestor. For a VM, you can `start`, `deploy`, `run`, `transfer`, and `terminate`. So, as you see, even the initial image deployment is sent and executed as a `script`, but the user does not need to bother about it, as it is all handled by the Golem SDKs.
```

Comment:
"depending on the time of type of the runtime utilized"

I think something is wrong here